SYCL/Reduction/reduction_complex_nums.cpp

```diff
@@ -0,0 +1,52 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <complex>
+#include <numeric>
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+// Currently, Identityless reduction for complex numbers is
+// only valid for plus operator.
+// TODO: Extend this test case once we support known_identity for std::complex
+// and more operators (apart from plus).
+template <typename T>
+void test_identityless_reduction_for_complex_nums(queue &q) {
+  // Allocate and initialize buffer on the host with all 1's.
+  buffer<std::complex<T>> valuesBuf { 1024 };
+  {
+    host_accessor a { valuesBuf };
+    std::fill(a.begin(), a.end(), std::complex<T>(1));
```

Comment:
I'd suggest to put something into the imaginary part as well. Also, maybe make the values non-uniform.